engines/darkseed/darkseed.cpp

```diff
@@ -0,0 +1,3746 @@
+			if (!bVar5) {
+				if (_inventory.hasObject(18)) {
+					bVar5 = true;
+				}
+			}
+			if (bVar5) {
+				isPlayingAnimation_maybe = false;
+				_objectVar.setObjectRunningCode(53, 1);
+				_player->loadAnimations("copcard.nsp");
+				animIndexTbl[0] = 0;
+				spriteAnimCountdownTimer[0] = 3;
+				_objectVar[1] = 2000;
+			}
+			else {
+				_console->addTextLine("The cops ignore your demands for attention.");
+			}
+		}
+		break;
+	case 65:
+		advanceAnimationFrame(0);
+		if (!_ObjRestarted) {
+			_player->_frameIdx = _player->_animations.getAnimAt(0).frameNo[animIndexTbl[0]];
+		} else {
+			_player->updateSprite();
+		}
+		break;
+	default:
+		error("Unhandled animation type! %d", otherNspAnimationType_maybe);
+	}
+}
+
+void DarkseedEngine::advanceAnimationFrame(int nspAminIdx) {
+	if (!_scaleSequence) {
+		scaledWalkSpeed_maybe = 1000;
+	}
+	else {
+		_room->calculateScaledSpriteDimensions(10, 10, _player->_position.y);
+	}
+	_ObjRestarted = false;
+	_FrameAdvanced = false;
+	const Obt &anim = _player->_animations.getAnimAt(nspAminIdx);
+	spriteAnimCountdownTimer[nspAminIdx] = spriteAnimCountdownTimer[nspAminIdx] - 1;
+	if (spriteAnimCountdownTimer[nspAminIdx] < 1) {
+		_FrameAdvanced = true;
+		animIndexTbl[nspAminIdx] = animIndexTbl[nspAminIdx] + 1;
+		_player->_position.x += ((int16)anim.deltaX[animIndexTbl[nspAminIdx]] * scaledWalkSpeed_maybe) / 1000;
+		_player->_position.y += ((int16)anim.deltaY[animIndexTbl[nspAminIdx]] * scaledWalkSpeed_maybe) / 1000;
+		if (animIndexTbl[nspAminIdx] == anim.numFrames) {
+			animIndexTbl[nspAminIdx] = 0;
+			_ObjRestarted = true;
+			isPlayingAnimation_maybe = false;
+			_player->_walkTarget.x = _player->_position.x;
+			_player->_walkTarget.y = _player->_position.y;
+			_player->updateSprite();
+		}
+		spriteAnimCountdownTimer[nspAminIdx] = anim.frameDuration[animIndexTbl[nspAminIdx]];
+	}
+}
+
+Common::Path DarkseedEngine::getRoomFilePath(const Common::Path &filename) {
+	if (isCdVersion()) {
+		return Common::Path("room").join(filename);
+	}
+	return filename;
+}
+
+Common::Path DarkseedEngine::getPictureFilePath(const Common::Path &filename) {
+	if (isCdVersion()) {
+		return Common::Path("picture").join(filename);
+	}
+	return filename;
+}
+
+void DarkseedEngine::handleObjCollision(int targetObjNum) {
+	if (targetObjNum == 35 && _objectVar[22] < 2 && _cursor.getY() > 40) {
+		targetObjNum = 22;
+	}
+	if (_actionMode == LookAction || _actionMode == HandAction || targetObjNum != 115) {
+		if (_cursor.getY() < 40 && _actionMode > LookAction) {
+			if (_objectVar[80] < 2 ||
+				((_actionMode != 25 || targetObjNum != 19) && (_actionMode != 19 || targetObjNum != 25))) {
+				if ((_actionMode == 25 && targetObjNum == 20) || (_actionMode == 20 && targetObjNum == 25)) {
+					if (_objectVar[25] > 99) { // Assemble special hammer
+						_console->printTosText(969);
+						_inventory.removeItem(20);
+						_inventory.removeItem(25);
+						updateDisplay();
+						playSound(17,5,-1);
```

Comment:
Same thing as the other places:
```suggestion
						playSound(17, 5, -1);
```